src/apify_client/consts.py

```diff
@@ -1,87 +1,54 @@
-    ACTOR_BUILD_FAILED = 'ACTOR.BUILD.FAILED'
-    #: The actor build has timed out
-    ACTOR_BUILD_TIMED_OUT = 'ACTOR.BUILD.TIMED_OUT'
-    #: The actor build was aborted
-    ACTOR_BUILD_ABORTED = 'ACTOR.BUILD.ABORTED'
-
-
-class MetaOrigin(str, Enum):
-    """Possible origins for actor runs, i.e. how were the jobs started."""
-
-    #: Job started from Developer console in Source section of actor
-    DEVELOPMENT = 'DEVELOPMENT'
-    #: Job started from other place on the website (either console or task detail page)
-    WEB = 'WEB'
-    #: Job started through API
-    API = 'API'
-    #: Job started through Scheduler
-    SCHEDULER = 'SCHEDULER'
-    #: Job started through test actor page
-    TEST = 'TEST'
-    #: Job started by the webhook
-    WEBHOOK = 'WEBHOOK'
-    #: Job started by another actor run
-    ACTOR = 'ACTOR'
+import warnings
+from typing import Any
+
+from apify_shared.consts import BOOL_ENV_VARS as BOOL_ENV_VARS_
+from apify_shared.consts import BOOL_ENV_VARS_TYPE as BOOL_ENV_VARS_TYPE_
+from apify_shared.consts import DATETIME_ENV_VARS as DATETIME_ENV_VARS_
+from apify_shared.consts import DATETIME_ENV_VARS_TYPE as DATETIME_ENV_VARS_TYPE_
+from apify_shared.consts import FLOAT_ENV_VARS as FLOAT_ENV_VARS_
+from apify_shared.consts import FLOAT_ENV_VARS_TYPE as FLOAT_ENV_VARS_TYPE_
+from apify_shared.consts import INTEGER_ENV_VARS as INTEGER_ENV_VARS_
+from apify_shared.consts import INTEGER_ENV_VARS_TYPE as INTEGER_ENV_VARS_TYPE_
+from apify_shared.consts import STRING_ENV_VARS as STRING_ENV_VARS_
+from apify_shared.consts import STRING_ENV_VARS_TYPE as STRING_ENV_VARS_TYPE_
+from apify_shared.consts import ActorEventTypes as ActorEventTypes_
+from apify_shared.consts import ActorExitCodes as ActorExitCodes_
+from apify_shared.consts import ActorJobStatus as ActorJobStatus_
+from apify_shared.consts import ActorSourceType as ActorSourceType_
+from apify_shared.consts import ApifyEnvVars as ApifyEnvVars_
+from apify_shared.consts import MetaOrigin as MetaOrigin_
+from apify_shared.consts import WebhookEventType as WebhookEventType_
+
+DEPRECATED_NAMES = [
+    'BOOL_ENV_VARS',
+    'BOOL_ENV_VARS_TYPE',
+    'DATETIME_ENV_VARS',
+    'DATETIME_ENV_VARS_TYPE',
+    'FLOAT_ENV_VARS',
+    'FLOAT_ENV_VARS_TYPE',
+    'INTEGER_ENV_VARS',
+    'INTEGER_ENV_VARS_TYPE',
+    'STRING_ENV_VARS',
+    'STRING_ENV_VARS_TYPE',
+    'ActorEventTypes',
+    'ActorExitCodes',
+    'ActorJobStatus',
+    'ActorSourceType',
+    'ApifyEnvVars',
+    'MetaOrigin',
+    'WebhookEventType',
+]
+
+
+def __getattr__(name: str) -> Any:
+    if name in DEPRECATED_NAMES:
+        warnings.warn(
+            (
+                f'Importing "{name}" from "apify_client.consts" is deprecated and will be removed in the future. '
+                'Please use "apify_shared" library instead.'
+            ),
+            category=DeprecationWarning,
+            stacklevel=2,
+        )
+        return globals()[f'{name}_']
```

Comment:
I think this won't work with any client-only constants we would define in here, it would raise `AttributeError` even if they would be present. Better to have some `else` branch for this.

Reply:
Actually, I took the structure from [PEP 562](https://peps.python.org/pep-0562/).

And it works, see the following example:
```python
import warnings
from typing import Any

from apify_shared.consts import ActorJobStatus as ActorJobStatus_
from apify_shared.consts import ActorSourceType as ActorSourceType_
from apify_shared.consts import MetaOrigin as MetaOrigin_
from apify_shared.consts import WebhookEventType as WebhookEventType_

DEPRECATED_NAMES = [
    'ActorJobStatus',
    'ActorSourceType',
    'MetaOrigin',
    'WebhookEventType',
]

A = 1


def __getattr__(name: str) -> Any:
    if name in DEPRECATED_NAMES:
        warnings.warn(
            (
                f'Importing "{name}" from "apify_client.consts" is deprecated and will be removed in the future. '
                'Please use "apify_shared" library instead.'
            ),
            category=DeprecationWarning,
            stacklevel=2,
        )
        return globals()[f'{name}_']
    raise AttributeError(f'module {__name__!r} has no attribute {name!r}')
```

Import deprecated const:
```python
>>> from src.apify_client.consts import ActorJobStatus
<stdin>:1: DeprecationWarning: Importing "ActorJobStatus" from "apify_client.consts" is deprecated and will be removed in the future. Please use "apify_shared" library instead.
>>> print(ActorJobStatus)
<enum 'ActorJobStatus'>
```

Import client specific const:
```python
>>> from src.apify_client.consts import A
>>> print(A)
1
```

Reply:
Ahaaaa, the custom `__getattr__()` is called only when the built-in `object.__getattribute__()` doesn't work, I understand it now. Kinda confusing tbh, but the performance rationale makes sense.

> If an attribute is not found on a module object through the normal lookup (i.e. `object.__getattribute__`), then `__getattr__` is searched in the module `__dict__` before raising an AttributeError. If found, it is called with the attribute name and the result is returned. Looking up a name as a module global will bypass module `__getattr__`. This is intentional, otherwise calling `__getattr__` for builtins will significantly harm performance.

Reply:
An explanatory note was added.